src/Common/src/Interop/Unix/System.Net.Security.Native/Interop.GssApi.cs

```diff
@@ -0,0 +1,125 @@
+            byte[] buffer,
+            out byte[] outputBuffer,
+            out uint outFlags)
+        {
+            outputBuffer = null;
+            outFlags = 0;
+
+            if (context == null)
+            {
+                context = new SafeGssContextHandle();
+            }
+
+            SafeGssBufferHandle outputToken;
+            NetSecurityNative.Status status;
+            NetSecurityNative.Status minorStatus;
+            int outputLength;
+            status = NetSecurityNative.InitSecContext(out minorStatus,
+                                                credential,
+                                                ref context,
+                                                isNtlm,
+                                                targetName,
+                                                (uint)inFlags,
+                                                buffer,
+                                                (buffer == null) ? 0 : buffer.Length,
+                                                out outputToken,
+                                                out outputLength,
+                                                out outFlags);
+
+            using (outputToken)
+            {
+                if ((status != NetSecurityNative.Status.GSS_S_COMPLETE) && (status != NetSecurityNative.Status.GSS_S_CONTINUE_NEEDED))
+                {
+                    throw new NetSecurityNative.GssApiException(SR.net_context_establishment_failed, status, minorStatus);
+                }
+
+                outputBuffer = new byte[outputLength];
+                outputToken.Copy(outputBuffer, 0);
+            }
+            return status == NetSecurityNative.Status.GSS_S_COMPLETE;
+        }
+
+        internal static byte[] Encrypt(
+            SafeGssContextHandle context,
+            bool encrypt,
+            byte[] buffer,
+            int offset,
+            int count)
+        {
+            Debug.Assert((buffer != null) && (buffer.Length > 0), "Invalid input buffer passed to Encrypt");
+            Debug.Assert((offset >= 0) && (offset < buffer.Length), "Invalid input offset passed to Encrypt");
+            Debug.Assert((count > 0) && (count <= (buffer.Length - offset)), "Invalid input count passed to Encrypt");
+
+            SafeGssBufferHandle outputToken;
+            int msgLength;
+            NetSecurityNative.Status minorStatus;
+            NetSecurityNative.Status status = NetSecurityNative.Wrap(out minorStatus, context, encrypt, buffer, offset, count, out outputToken, out msgLength);
+            using (outputToken)
+            {
+                if (status != NetSecurityNative.Status.GSS_S_COMPLETE)
+                {
+                    throw new NetSecurityNative.GssApiException(SR.net_context_wrap_failed, status, minorStatus);
+                }
+
+                byte[] outputBuffer = new byte[msgLength];
+                outputToken.Copy(outputBuffer, 0);
+                return outputBuffer;
```

Comment:
Thus far, all of the places I've seen us get an outputToken then result in copying it to a byte[] and disposing of the outputToken.  Is there a better way of factoring this that wouldn't result in the intermediate token being created and disposed, that would consolidate the boilerplate, etc.?